mobile/lib/features/trade/submit_order_change_notifier.dart

```diff
@@ -16,16 +18,17 @@ class PendingOrder {
 }
 
 class SubmitOrderChangeNotifier extends ChangeNotifier {
+  final OrderService orderService = OrderService();
+
   PendingOrder? _pendingOrder;
 
   submitPendingOrder(TradeValues tradeValues) async {
     _pendingOrder = PendingOrder(tradeValues);
 
     notifyListeners();
 
-    // TODO: Actually submit the pending order...
-    await Future.delayed(const Duration(seconds: 1));
-    // TODO: Handle submit failure
+    await orderService.submitMarketOrder(
+        tradeValues.leverage, tradeValues.quantity, ContractSymbol.btcusd, tradeValues.direction);
     _pendingOrder!.state = PendingOrderState.submittedSuccessfully;
 
     notifyListeners();
```

Comment:
Yes, because the model changes - when we set the `PendingOrder` it is in status `pending` and once we have a reply from the api call we update the status.

First, we notify that a `PendingOrder` exists, then we notify about the status change after the call to the backend.
I an try to add another function to make it more clear :)

Reply:
I opted for adding comments